packages/typespec-mcp/lib/main.tsp

```diff
@@ -31,6 +31,11 @@ extern dec idempotent(target: Reflection.Operation, value?: boolean);
  */
 extern dec closedWorld(target: Reflection.Operation, value?: boolean);
 
+/**
+ * Declare an operation to be a MCP Resource.
+ */
+extern dec resource(target: Reflection.Operation);
```

Comment:
I think we need a few more things for resources. 
As far as I understand resource can either be static or templates but either way it needs some uri.

Should the API be something like that

```tsp
@resource("config://app")
op config(): Resource;

@resource("users://{userId}/profile")
op profile(userId: string): Resource;
```